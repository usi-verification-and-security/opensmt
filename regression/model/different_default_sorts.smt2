(set-option :print-success false)
(set-option :produce-models true)
(set-logic QF_UF)
(declare-sort x4 0)
(declare-sort x1 0)
(declare-fun x2 () x1)
(declare-fun x6 (x4) x1)
(declare-fun x3 (x1) x4)
(declare-fun x5 () x4)
(assert (= (x3 x2) x5))
(assert (= (x6 x5) x2))
(check-sat)
(get-model)
