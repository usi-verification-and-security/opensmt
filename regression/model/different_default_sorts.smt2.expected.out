sat
(
  (define-fun x2 () x1
    (as @2 x1))
  (define-fun x6 ((r0 x4)) x1
    (ite (= (as @3 x4) r0) (as @2 x1) (as @d3 x1)))
  (define-fun x3 ((r1 x1)) x4
    (ite (= (as @2 x1) r1) (as @3 x4) (as @d2 x4)))
  (define-fun x5 () x4
    (as @3 x4))
)
